### ISSUE TITLE
Toggle Focus in 2022 -- switching between track pad and keyboard

### ISSUE THREAD

**Issue by lansingthomas** [atlas-engineer/nyxt#2706]
Please let me know if this issue is discussed elsewhere or if a solution exists. I think the feature is important for usability and I have been unable to find discussion about it. I provide one example but I suspect this may apply to many Nyxt features.

## Context / problem
- as a navigator and reader in the browser, I find myself interacting with web pages. Often using my trackpad to scroll--> When it comes time to use another Nyxt input I find myself hitting the keys to no effect.
- often this issue arises when I want to close the Nyxt window with `esc` to keep reading a one page website, or to uncover information.
  - Example 1:
     - while reading a long article, I input `Ctrl+j` intending to interact with a link using the _follow-hints_  feature .--> 
     - link is now covered by the Nyxt window and I am uncertain which hint I need --> 
     - I `move hands to trackpad` --> (rendering the _follow-hints_ feature meaningless for navigation purposes)
     - I `click the page with trackpad` to enable trackpad scrolling to view the obstructed content --> 
     - I `scroll with trackpad` (link is now visible again - including the hint) --> 
     - at this point I WANT to enter the proper code (`AGP` or similar) in the Nyxt window to follow the now unobstructed link.  
     - the ERROR is that any keys I press at this point are inputing to the webpage, not the Nyxt window.

The current way to avoid moving my hands `keyboard` -> `trackpad` -> `keyboard` -> `trackpad` -> `keyboard` is to always remember that the Nyxt window will force me in this small way. This is cognitive load for users, which is bad, so lets help : )

**Same issue in another context**
- I often also feel the pain of this issue when I want to close a window. Perhaps when I start to execute a command then change my mind while clicking around.
   - I often find myself hitting `esc`--> no response (because my buffer input is set to the webpage).

## please take me here
- understanding that I can click the Nyxt window to resolve this, I really want to switch the focus of my keyboard inputs with a hotkey to avoid moving my hand back to the trackpad (just to enable using the keyboard again).
- Perhaps the first press of `esc` could global binding switch my keyboard-input focus back to the Nyxt window. This way I could double tap `esc` to close any Nyxt window.

**Comment by aadcg** [comment]
@lansingthomas lots of topics to unbundle here!

> link is now covered by the Nyxt window and I am uncertain which hint I need

It's possible to configure that. `C-SPC describe-slot RET fit-to-prompt-p RET` and then hit the `Configure` button. This might still sound a bit hard to follow so I will introduce you to Nyxt configuration in our next session.

It seems to me that you find that a UI that doesn't cover part of the web page is the most intuitive. I have been singing that song for a while...

> I click the page with trackpad to enable trackpad scrolling to view the obstructed content

You don't need to do this actually. We have keybinding bound for those operations but, since you're using the `cua` keyscheme, they're unbound. More gibberish information. I assure you that I'm not randomly connecting all of these words!

 
Instead of me continuing to comment over your issue, let's talk about this next Friday. Everything will become clearer.

**Comment by lansingthomas** [comment]
I am closing this because I found a better way to explain the issue. See #2464

**Comment by aadcg** [comment]
There might still be topics worth exploring here. I haven't yet explained lots of things that you'll be happy to know about :)

Think which of these observations apply to the prompt buffer, so that we can enrich the specification for #2687.
